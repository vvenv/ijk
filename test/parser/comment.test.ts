import { describe, expect, test } from 'vitest';
import { parse } from './__helper';

describe('stripComments', () => {
  describe('on', () => {
    test('basic', () => {
      expect(parse('{{# foo #}}')).toMatchSnapshot();
      expect(parse('{{# foo #}}')).toMatchSnapshot();
    });

    test('multiline', () => {
      expect(parse('{{# foo\nbar #}}')).toMatchSnapshot();
    });
  });

  describe('off', () => {
    test('basic', () => {
      expect(parse('{{# foo #}}', { stripComments: false })).toMatchSnapshot();
      expect(parse('{{# foo #}}', { stripComments: false })).toMatchSnapshot();
    });

    test('multiline', () => {
      expect(
        parse('{{# foo\nbar #}}', { stripComments: false }),
      ).toMatchSnapshot();
    });
  });
});
