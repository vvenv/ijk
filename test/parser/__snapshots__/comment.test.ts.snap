// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`stripComments > off > basic 1`] = `
{
  "out": Out {
    "content": ""use strict";let s = "";s+="<!-- foo -->";return s;",
    "options": {
      "autoEscape": true,
      "debug": false,
      "strictMode": true,
      "stripComments": false,
      "tagEnd": "}}",
      "tagStart": "{{",
      "trimWhitespace": true,
    },
  },
  "smp": SMP {
    "options": {
      "autoEscape": true,
      "debug": false,
      "strictMode": true,
      "stripComments": false,
      "tagEnd": "}}",
      "tagStart": "{{",
      "trimWhitespace": true,
    },
    "template": "{{# foo #}}",
  },
}
`;

exports[`stripComments > off > basic 2`] = `
{
  "out": Out {
    "content": ""use strict";let s = "";s+="<!-- foo -->";return s;",
    "options": {
      "autoEscape": true,
      "debug": false,
      "strictMode": true,
      "stripComments": false,
      "tagEnd": "}}",
      "tagStart": "{{",
      "trimWhitespace": true,
    },
  },
  "smp": SMP {
    "options": {
      "autoEscape": true,
      "debug": false,
      "strictMode": true,
      "stripComments": false,
      "tagEnd": "}}",
      "tagStart": "{{",
      "trimWhitespace": true,
    },
    "template": "{{# foo #}}",
  },
}
`;

exports[`stripComments > off > multiline 1`] = `
{
  "out": Out {
    "content": ""use strict";let s = "";s+="<!-- foo bar -->";return s;",
    "options": {
      "autoEscape": true,
      "debug": false,
      "strictMode": true,
      "stripComments": false,
      "tagEnd": "}}",
      "tagStart": "{{",
      "trimWhitespace": true,
    },
  },
  "smp": SMP {
    "options": {
      "autoEscape": true,
      "debug": false,
      "strictMode": true,
      "stripComments": false,
      "tagEnd": "}}",
      "tagStart": "{{",
      "trimWhitespace": true,
    },
    "template": "{{# foo
bar #}}",
  },
}
`;

exports[`stripComments > on > basic 1`] = `
{
  "out": Out {
    "content": ""use strict";let s = "";return s;",
    "options": {
      "autoEscape": true,
      "debug": false,
      "strictMode": true,
      "stripComments": true,
      "tagEnd": "}}",
      "tagStart": "{{",
      "trimWhitespace": true,
    },
  },
  "smp": SMP {
    "options": {
      "autoEscape": true,
      "debug": false,
      "strictMode": true,
      "stripComments": true,
      "tagEnd": "}}",
      "tagStart": "{{",
      "trimWhitespace": true,
    },
    "template": "{{# foo #}}",
  },
}
`;

exports[`stripComments > on > basic 2`] = `
{
  "out": Out {
    "content": ""use strict";let s = "";return s;",
    "options": {
      "autoEscape": true,
      "debug": false,
      "strictMode": true,
      "stripComments": true,
      "tagEnd": "}}",
      "tagStart": "{{",
      "trimWhitespace": true,
    },
  },
  "smp": SMP {
    "options": {
      "autoEscape": true,
      "debug": false,
      "strictMode": true,
      "stripComments": true,
      "tagEnd": "}}",
      "tagStart": "{{",
      "trimWhitespace": true,
    },
    "template": "{{# foo #}}",
  },
}
`;

exports[`stripComments > on > multiline 1`] = `
{
  "out": Out {
    "content": ""use strict";let s = "";return s;",
    "options": {
      "autoEscape": true,
      "debug": false,
      "strictMode": true,
      "stripComments": true,
      "tagEnd": "}}",
      "tagStart": "{{",
      "trimWhitespace": true,
    },
  },
  "smp": SMP {
    "options": {
      "autoEscape": true,
      "debug": false,
      "strictMode": true,
      "stripComments": true,
      "tagEnd": "}}",
      "tagStart": "{{",
      "trimWhitespace": true,
    },
    "template": "{{# foo
bar #}}",
  },
}
`;
